### PR TITLE
fix(IDX): add branch head sha

### PR DESCRIPTION
**Pull request by cgundy**
Fixes a small bug with identifying the diff files for `diff.sh`.

### CODE REVIEW

**Review by basvandijk — APPROVED** [review]
Yes, this does seem more correct. It did seem to work before right? Can you point to a run where this resulted in something unexpected?

**Comment by cgundy** [comment]
Yes actually just my previous PR: https://github.com/dfinity/ic/actions/runs/11160256899/job/31020363515 - it was taking in diffs for way more files than were in my PR.

**Comment by cgundy** [comment]
I think this is where the issue was introduced: https://github.com/dfinity/ic/pull/1698/files

We were doing the same thing, just pulling the HEAD of the target branch. But let me take another look at this to make sure we handle cases where those variables might not be set.

File: ci/bazel-scripts/diff.sh [hunk 1]
```diff
@@ -14,7 +14,8 @@ set -x
 cd "$(git rev-parse --show-toplevel)"
 
 MERGE_BASE="${MERGE_BASE_SHA:-HEAD}"
-COMMIT_RANGE=${COMMIT_RANGE:-$MERGE_BASE".."}
+# we can't use HEAD here because that is the merge commit which contains the changes of the current HEAD of master
+COMMIT_RANGE=$MERGE_BASE".."${BRANCH_HEAD_SHA:-}
```
Comment (basvandijk) [inline ci/bazel-scripts/diff.sh:18]: This would be safer:
```suggestion
COMMIT_RANGE="$MERGE_BASE..${BRANCH_HEAD_SHA:-}"
```

**Pull request merged**